plugins/catalog-backend/src/service/DefaultEntitiesCatalog.ts

```diff
@@ -392,44 +397,26 @@ export class DefaultEntitiesCatalog implements EntitiesCatalog {
   }
 
   async facets(request: EntityFacetsRequest): Promise<EntityFacetsResponse> {
-    const { entities } = await this.entities({
-      filter: request.filter,
-      authorizationToken: request.authorizationToken,
-    });
-
     const facets: EntityFacetsResponse['facets'] = {};
+    const db = this.database;
 
     for (const facet of request.facets) {
-      const values = entities
-        .map(entity => {
-          // TODO(freben): Generalize this code to handle any field that may
-          // have dots in its key?
-          if (facet.startsWith('metadata.annotations.')) {
-            return entity.metadata.annotations?.[
-              facet.substring('metadata.annotations.'.length)
-            ];
-          } else if (facet.startsWith('metadata.labels.')) {
-            return entity.metadata.labels?.[
-              facet.substring('metadata.labels.'.length)
-            ];
-          }
-          return lodash.get(entity, facet);
-        })
-        .flatMap(field => {
-          if (typeof field === 'string') {
-            return [field];
-          } else if (Array.isArray(field)) {
-            return field.filter(i => typeof i === 'string');
-          }
-          return [];
-        })
-        .sort();
+      const dbQuery = db<DbSearchRow>('search')
+        .join('final_entities', 'search.entity_id', 'final_entities.entity_id')
+        .where('search.key', facet.toLocaleLowerCase('en-US'))
+        .count('search.entity_id as count')
```

Comment:
```suggestion
        .count('search.entity_id', { as: 'count' })
```
Afaik that's the correct usage